### PR TITLE
Consistensy checks added to AUP importer

### PR DESCRIPTION
**Pull request by vsverchinsky**
Resolves: #5211
Resolves: #5304

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

QA:  Might want to wait until this is rebased onto #5260 
   - [x] Save 2.4.2 projects with misaligned stereo, mismatched channel formats, mismatched channel rates
   - [x] Open them in 3.4, seeing similar warnings and same splitting, as when opening 3.x legacy projects with similar
   - [ ] 2.4.2 projects without those problems open with no warnings

### CODE REVIEW

File: modules/mod-aup/ImportAUP.cpp
Comment (Paul-Licameli) [inline modules/mod-aup/ImportAUP.cpp:560]: Commit comment:  "consistency"

File: modules/mod-aup/ImportAUP.cpp
Comment (Paul-Licameli) [inline modules/mod-aup/ImportAUP.cpp:574]: I'm convinced that this is correct to repair the track consistency, but there is also new logic in `ProjectFileManager::ReadProjectFile` to warn the user about the change, and that function is not reached.

It should not be difficult to make that reusable.

**Review by Paul-Licameli — DISMISSED** [review]
Should give the user information about fixes, just as if upgrading an .aup3 project from 3.x

File: modules/mod-aup/ImportAUP.cpp
Comment (vsverchinsky) [reply modules/mod-aup/ImportAUP.cpp:574]: My guess that the main reason to show this dialog is to warn about possible effect stack loss, when you import AUP with mismatched channels nothing gets removed

Comment (Paul-Licameli) [reply modules/mod-aup/ImportAUP.cpp:574]: But you are also changing it in #5260 to give other warnings about mismatched rates and formats

Comment (vsverchinsky) [reply modules/mod-aup/ImportAUP.cpp:574]: The warning is about effect stacks that can be removed as we split channels, we don't change sample format or rate

Comment (vsverchinsky) [reply modules/mod-aup/ImportAUP.cpp:574]: And existing texts do not fit in that case I think

Comment (LWinterberg) [reply modules/mod-aup/ImportAUP.cpp:574]: the warning contains two pieces of information: 
1. your project will look different to what you last saved because we migrated some stereo tracks to mono tracks
2. realtime effect stacks may be lost. 

In the case of legacy importers, part 2 is not relevant but part 1 is, so I'd still want to see the message box.

File: src/tracks/playabletrack/wavetrack/WaveTrackUtils.cpp
Comment (Paul-Licameli) [inline src/tracks/playabletrack/wavetrack/WaveTrackUtils.cpp:32]: Strings, @LWinterberg  ?

This function causes only the first applicable message to be shown.  Possibly more that one might apply in an old project.  Maybe all applicable messages should be shown.  Or maybe we don't care about this edge case of an edge case.

File: src/ProjectFileManager.cpp
Comment (Paul-Licameli) [inline src/ProjectFileManager.cpp:201]: However, I see that in case of opening a 3.x file, not importing .aup, the warning about possible loss of stacked effects is always given, in addition to whatever other message is reported by the function.  That is unchanged behavior compared with master.

File: modules/mod-aup/ImportAUP.cpp
Comment (Paul-Licameli) [inline modules/mod-aup/ImportAUP.cpp:361]: Delete the second line of the comment too

File: modules/mod-aup/ImportAUP.cpp
Comment (Paul-Licameli) [inline modules/mod-aup/ImportAUP.cpp:454]: This looks correct but it's all duplication of a part of `ProjectFileManager::ReadProjectFile`.  Can there be just one function to do this?

**Comment by Paul-Licameli** [comment]
I dismissed my request for changes, but before approval I will await an answer from @LWinterberg about a new question, and maybe @vsverchinsky will eliminate duplicated logic.

Also, correct the spelling in the commit comment please.

File: src/tracks/playabletrack/wavetrack/WaveTrackUtils.cpp
Comment (LWinterberg) [reply src/tracks/playabletrack/wavetrack/WaveTrackUtils.cpp:32]: strings look good!

**Comment by vsverchinsky** [comment]
Corrected the spelling and duplicated logic extracted.
Also loop that fixes consistensy has changed a bit - now it continues to remove effect stacks from unlinked tracks even if error has been found.

**Review by Paul-Licameli — APPROVED** [review]
Good to see elimination of duplicated logic

**Pull request merged**
